sys/mips/pmap.c

```diff
@@ -307,6 +308,31 @@ static bool pmap_extract_nolock(pmap_t *pmap, vaddr_t va, paddr_t *pap) {
   return true;
 }
 
+bool pmap_extract_and_hold(pmap_t *pmap, vaddr_t va, paddr_t *pap,
+                           vm_prot_t prot) {
+  assert(pap != NULL);
+  paddr_t old = *pap;
```

Comment:
Do we really need to remember old value of `*pap`?
`pmap_extract_and_hold` is a kernel internal function so you don't give any information to user - if you could call `pmap_extract_and_hold` you can also call `pmap_extract`. We already give a information if `*pap` as valid by returning `true` or `false` from that function.

Reply:
ok. I changed it.